src/commands/schedule.ts

```diff
@@ -24,149 +22,110 @@ command.addStringOption((option) =>
-      });
-
-      // TODO if I am able ot use the query builder then I can use types for the messaging, but video_participants it not a column?
-
-      // const videos: Video[] = await AppDataSource.getRepository(Video)
-      //   .createQueryBuilder("videos")
-      //   .select("videos.id")
-      //   .distinct(true)
-      //   .leftJoin("video_participants", "video_participants")
-      //   .where("videos.scheduledTime > datetime('now')")
-      //   .andWhere("videos.hostStreamer = :streamer OR video_participants = :streamer", {
-      //     streamer: streamer,
-      //   })
-      //   .orderBy("videos.scheduledTime", "ASC")
-      //   .getMany();
 
-      // const videos = (await AppDataSource.getRepository(Video)
-      //   .createQueryBuilder("videos")
-      //   .distinctOn(["videos.id"])
-      //   .select("videos.id")
-      //   .select("videos.title")
-      //   .select("videos.scheduled_time", "scheduledTime")
-      //   .select("videos.host_streamer_id", "hostStreamerId")
-      //   .leftJoin("video_participants", "vp", "video_participants.video_id = videos.id")
-      //   .andWhere("videos.scheduled_time > datetime('now')")
-      //   .andWhere("videos.scheduled_time < datetime('now', '+10 day')")
-      //   .andWhere((q) =>
-      //     q
-      //       .orWhere("videos.host_streamer_id = ?", [streamer.id])
-      //       .orWhere("video_participants.participant_streamer_id = ?", [streamer.id])
-      //   )
-      //   .orderBy("videos.scheduled_time", "ASC")
-      //   .execute()) as VideoData[];
-
-      const videos = (await AppDataSource.query(
-        `
-      SELECT DISTINCT videos.id, videos.title, videos.scheduled_time "scheduledTime", videos.host_streamer_id "hostStreamerId"
-      FROM videos
-      LEFT JOIN video_participants ON video_participants.video_id = videos.id
-      WHERE
-        videos.scheduled_time > datetime('now')
-        AND videos.scheduled_time < datetime('now', '+10 day')
-        AND (
-          videos.host_streamer_id = ?
-          OR video_participants.participant_streamer_id = ?
+      await embedScheduleFormatter(videos, interaction);
+      return;
+    } else if (streamerId !== null) {
+      // show schedule for specific streamer
+      const videos = await AppDataSource.getRepository(Video)
+        .createQueryBuilder("videos")
+        .where("videos.scheduledTime > :date", { date: new Date() })
+        .andWhere(
+          new Brackets((qb) => {
+            qb.where("videos.hostStreamer.id = :streamerId", { streamerId }).orWhere(
+              "videos.id IN (SELECT vp.video_id FROM video_participants vp WHERE vp.participant_streamer_id = :streamerId)",
+              { streamerId }
+            );
+          })
         )
-      ORDER BY videos.scheduled_time ASC
-      `,
-        [streamer.id, streamer.id]
-      )) as VideoData[];
+        .leftJoinAndSelect("videos.participantStreamers", "ps")
+        .leftJoinAndSelect("videos.hostStreamer", "hs")
```

Comment:
このSQLの結果例を送ってくれませんか。`SELECT FROM videos`なら`LEFT JOIN videos`の目標がよくわかりません

Reply:
These are essentially to load the selections from above (kind of like eager loading) for the `videos`. 
I'm not sure why but doing `leftJoinAndSelect` earlier doesn't work... and regardless if it worked for one I would have to put them same concept here for the other (because if I do 2 selects then it'll overwrite)

Reply:
よくわかりませんが🐍にしないといけなさそう